tests/config/test_component_factory.py

```diff
@@ -0,0 +1,32 @@
+from pathlib import Path
+from typing import Dict, Union
+
+from omegaconf import OmegaConf
+
+from modalities.config.component_factory import ComponentFactory
+from tests.config.hierarchical_instantiation.configs import (
+    CompVConfig,
+    CompWConfig,
+    CompXConfig,
+    CompYConfig,
+    CompZConfig,
+    ReferenceConfig,
+)
+
+
+def load_app_config_dict(config_file_path: Path) -> Dict:
+    cfg = OmegaConf.load(config_file_path)
+    config_dict = OmegaConf.to_container(cfg, resolve=True)
+    return config_dict
+
+
+config_file_path = Path("config.yaml")
```

Comment:
This relative path should be replaced by an absolute path. Otherwise, `pytest` will only work if executed from the file's directory, not the root directory. A solution could be to use `_ROOT_DIR` as in `tests/conftest.py`